server/routes/refer.ts

```diff
@@ -22,40 +22,133 @@ export default function routes(controllers: Controllers, services: Services, rou
     oasysConfirmationController,
   } = controllers
 
-  get(referPaths.start.pattern, referralsController.start())
-  get(referPaths.new.pattern, referralsController.new())
+  get(referPaths.start.pattern, referralsController.start(), { auditEvent: 'REFER_VIEW_START' })
+  get(referPaths.new.pattern, referralsController.new(), { auditEvent: 'REFER_VIEW_PERSON_SEARCH' })
 
-  post(referPaths.people.find.pattern, peopleController.find())
-  get(referPaths.people.show.pattern, peopleController.show())
+  post(referPaths.people.find.pattern, peopleController.find(), {
+    auditBodyParams: ['prisonNumber'],
+    auditEvent: 'REFER_SEARCH_PERSON',
+  })
+  get(referPaths.people.show.pattern, peopleController.show(), {
+    auditEvent: 'REFER_VIEW_FOUND_PERSON',
+  })
 
-  post(referPaths.create.pattern, referralsController.create())
-  get(referPaths.show.pattern, referralsController.show())
-  get(referPaths.showPerson.pattern, referralsController.showPerson())
+  post(referPaths.create.pattern, referralsController.create(), { auditEvent: 'REFER_CREATE_DRAFT_REFERRAL' })
+  get(referPaths.show.pattern, referralsController.show(), { auditEvent: 'REFER_VIEW_TASKLIST' })
+  get(referPaths.showPerson.pattern, referralsController.showPerson(), {
+    auditBodyParams: ['prisonNumber'],
+    auditEvent: 'REFER_VIEW_REFERRED_PERSON',
+  })
 
-  get(referPaths.confirmOasys.show.pattern, oasysConfirmationController.show())
-  put(referPaths.confirmOasys.update.pattern, oasysConfirmationController.update())
+  get(referPaths.confirmOasys.show.pattern, oasysConfirmationController.show(), {
+    auditEvent: 'REFER_VIEW_CONFIRM_OASYS',
+  })
+  put(referPaths.confirmOasys.update.pattern, oasysConfirmationController.update(), {
+    redirectAuditEventSpecs: [
```

Comment:
I get what this is doing but `redirectAuditEventSpecs` is a bit of a brain melting combination of words 😅

Also, will we get a test failure if we add a new redirect or it redirects somewhere unexpected? I can see us forgetting to update this if adding/changing redirects in existing pages,